presto-tests/src/main/java/com/facebook/presto/tests/statistics/MetricComparisonStrategies.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.tests.statistics;
+
+import com.google.common.collect.Range;
+
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class MetricComparisonStrategies
+{
+    private MetricComparisonStrategies() {}
+
+    public static MetricComparisonStrategy noError()
+    {
+        return absoluteError(0);
+    }
+
+    public static MetricComparisonStrategy absoluteError(double error)
+    {
+        return absoluteError(Range.closed(-error, error));
+    }
+
+    public static MetricComparisonStrategy absoluteError(Range<Double> errorRange)
+    {
+        return (actual, estimate) -> mapRange(errorRange, endpoint -> endpoint + actual)
+                .contains(estimate);
```

Comment:
I can imagine writing this in a less readable way, but admittedly, it would not be easy.

Alas, is it obvious from the code what happens if any of the 4 compared values (`actual`, `estimate`, `errorRange`'s ends) is infinity or NaN?

Reply:
`less readable way` I understand that you wanted to say `more sophisticated way` :)